laa_crime_application_store_app/services/v1/notification_service.py

```diff
@@ -0,0 +1,64 @@
+from uuid import UUID
+
+import httpx
+import structlog
+from fastapi import Request
+from sqlalchemy.orm import Session
+
+from laa_crime_application_store_app.models.subscriber_schema import Subscriber
+
+logger = structlog.getLogger(__name__)
+
+
+class NotificationService:
+    @staticmethod
+    def subscribe(db: Session, subscriber_type: str, webhook_url: str):
+        existing_subscriber = (
+            db.query(Subscriber).filter(Subscriber.webhook_url == webhook_url).first()
+        )
+        if existing_subscriber is not None:
+            return None
+
+        new_subscriber = Subscriber(
+            subscriber_type=subscriber_type,
+            webhook_url=webhook_url,
+        )
+        db.add(new_subscriber)
+        db.commit()
+
+        return new_subscriber
+
+    @staticmethod
+    def unsubscribe(db: Session, subscriber_type: str, webhook_url: str):
+        existing_subscriber = (
+            db.query(Subscriber)
+            .filter(
+                Subscriber.webhook_url == webhook_url,
+                Subscriber.subscriber_type == subscriber_type,
+            )
+            .first()
+        )
+        if existing_subscriber is None:
+            return False
+
+        db.delete(existing_subscriber)
```

Comment:
wrap in try except

Reply:
Thinking about it, the only way this could fail would be if between retrieving the record on line 40 and deleting it here, the record was deleted by another process. So rather than try/except to fail gracefully in that case, I've added `with_for_update`